src/core/thread/mle.cpp

```diff
@@ -1424,6 +1443,14 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
             continue;
         }
 
+#if OPENTHREAD_CONFIG_TMF_DUA_REGISTRATION_ENABLE
+        // Here skips DUA which was appended already.
+        if (addr->GetAddress() == GetDomainUnicastAddress())
```

Comment:
Better just compare by pointer address?
```suggestion
        if (&(addr->GetAddress()) == &GetDomainUnicastAddress())
```